src/cargo-kani/src/args_toml.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::{bail, Result};
+use std::collections::BTreeMap;
+use std::ffi::OsString;
+use toml::Value;
+
+/// Produces a set of arguments to pass to "self" (cargo-kani) from a Cargo.toml project file
+pub fn config_toml_to_args() -> Result<Vec<OsString>> {
+    // TODO: `cargo locate-project` maybe?
+    let file = std::fs::read_to_string("Cargo.toml");
+    if let Ok(file) = file {
+        toml_to_args(&file)
+    } else {
+        // Suppress the error if we can't find it, for now.
+        Ok(vec![])
+    }
+}
+
+/// Parse a config toml and extract the cargo-kani arguments we should try injecting
+fn toml_to_args(file: &str) -> Result<Vec<OsString>> {
+    let config = file.parse::<Value>()?;
+    // We specifically rely on BTreeMap here to get a stable iteration order for unit testing
+    let mut map: BTreeMap<String, Value> = BTreeMap::new();
+    let keys = ["workspace.metadata.kani.flags", "package.metadata.kani.flags", "kani.flags"];
+
+    for key in keys {
+        if let Some(val) = descend(&config, key) {
+            if let Some(val) = val.as_table() {
+                map.extend(val.iter().map(|(x, y)| (x.to_owned(), y.to_owned())));
+            }
+        }
+    }
+
+    let mut args = Vec::new();
+    let mut suffixed_args = Vec::new();
+
+    for (flag, value) in map {
+        if flag == "cbmc-args" {
+            insert_arg_from_toml(&flag, &value, &mut suffixed_args)?;
+        } else {
+            insert_arg_from_toml(&flag, &value, &mut args)?;
+        }
+    }
+
+    args.extend(suffixed_args);
+
+    Ok(args)
+}
+
+/// Translates one toml entry into arguments and inserts it into args
+fn insert_arg_from_toml(flag: &str, value: &Value, args: &mut Vec<OsString>) -> Result<()> {
```

Comment:
It should be the same thing as happens on the command line so:

```
$ cargo-kani --verbose --verbose
error: The argument '--verbose' was provided more than once, but cannot be used multiple times

USAGE:
    cargo-kani --abs-type <abs-type> --function <function> --object-bits <object-bits> --output-format <output-format> --verbose

For more information try --help
```

Reply:
The command line argument should take precedence over the cargo configuration though. Like it does today. I believe this works today because the python implementation allows you to use an argument more than once.

On clap you can set the following setting [AllArgsOverrideSelf](https://docs.rs/clap/2.33.0/clap/enum.AppSettings.html) to enable that behavior.